src/main/java/com/javadeobfuscator/deobfuscator/transformers/smoke/StringEncryptionTransformer.java

```diff
@@ -66,13 +67,47 @@ public void transform() throws Throwable {
         							MethodNode decrypterNode = innerClassNode.methods.stream().filter(mn -> mn.name.equals(m.name) && mn.desc.equals(m.desc)).findFirst().orElse(null);
         							if(isSmokeMethod(decrypterNode))
         							{
+        								context.push(wrappedClassNode.classNode.name, methodNode.name, wrappedClassNode.constantPoolSize);
 	        							String value = MethodExecutor.execute(wrappedClassNode, decrypterNode, Arrays.asList(JavaValue.valueOf(obfString), new JavaInteger(number)), null, context);
 	                                    methodNode.instructions.remove(m.getPrevious().getPrevious());
 	                                    methodNode.instructions.remove(m.getPrevious());
 	                                    methodNode.instructions.set(m, new LdcInsnNode(value));
 	                                    count.getAndIncrement();
         							}
         						}
+        					}else
```

Comment:
The new `ConstantFolder` inside peephole should handle this (and even if it doesn't right now, it should be). Dash-O's 'advanced' encryption used to fail because of something similar. Do you have a sample of Smoke that I can take a look at?

Reply:
http://dl.liquidbounce.net/download/CCBlueX/LiquidBounce/1.8/a15d8sdf41/LiquidBounce1.8.9-b26.2.zip

Reply:
I don't understand. What do you mean by you created?

Reply:
(It was just a note that I do heavily doing coding on this deobfuscator just for the fun)

Reply:
ah, I see